src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,143 @@
+        indexedElement.prev = newElement;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getIndexedElement(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        T oldItem = getIndexedElement(index).item;
+        getIndexedElement(index).item = value;
+        return oldItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> deletedNode = getIndexedElement(index);
+        unlink(deletedNode);
+        return deletedNode.item;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> deletedNode;
+        for (int i = 0; i < size; i++) {
+            deletedNode = getIndexedElement(i);
+            if ((deletedNode.item != null && deletedNode.item.equals(object))
+                    || (deletedNode.item == null && object == null)) {
+                unlink(deletedNode);
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node<T> getIndexedElement(int index) {
+        if ((index >= size || index < 0) && index != 0) {
+            throw new IndexOutOfBoundsException("Index " + index + " doesn`t exist."
+                    + " Actual size is " + size);
+        }
+        Node<T> indexedElement;
+        if (index < size / 2) {
+            indexedElement = head;
+            for (int i = 0; i < index; i++) {
+                indexedElement = indexedElement.next;
+            }
+        } else {
+            indexedElement = tail;
+            for (int i = 0; i < size - index - 1; i++) {
+                indexedElement = indexedElement.prev;
+            }
+        }
+        return indexedElement;
+    }
+
+    private void unlink(Node<T> node) {
+        if (size-- == 1) {
+            head = null;
+            tail = null;
+            return;
+        }
+        if (node == head) {
+            node.next.prev = null;
+            head = node.next;
+            return;
+        } else if (node == tail) {
+            node.prev.next = null;
+            tail = node.prev;
+            return;
+        }
+        node.prev.next = node.next;
+        node.next.prev = node.prev;
+    }
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            return super.equals(obj);
+        }
```

Comment:
we don't need equals here, and what's the point to override it if you leave the same implementation

Reply:
oops i forgot to delete that